tests/ci/test_test_quality_guardrails.py

```
@@ -68,7 +69,27 @@ def _git_changed_test_files() -> list[Path]:
     suite.  Only files touched in the current branch are checked, preventing
     failures on historical code while blocking new violations.
+
+    In CI, a shallow checkout can leave ``origin/main`` unresolvable, which
+    would otherwise silently collapse the diff to empty and skip every
+    violation. Fail loudly in that case rather than degrade to a no-op
+    guardrail. Outside CI (local dev, possibly without a configured
+    ``origin`` remote) the silent fallback below is intentional.
     """
+    if os.environ.get("GITHUB_ACTIONS") == "true":
+        verify = subprocess.run(
+            ["git", "rev-parse", "--verify", "--quiet", "origin/main"],
```

Ruff (0.15.18)

[error] Starting a process with a partial executable path

(S607)